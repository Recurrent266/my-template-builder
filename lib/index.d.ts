export * from './kfintech-typescript-app-project';
export * from './kfintech-nestjs-app-project';
export * from './utils/replace-placeholders';
