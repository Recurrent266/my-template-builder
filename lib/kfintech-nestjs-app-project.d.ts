import { KFinTechTypeScriptAppProject, KFinTechTypeScriptAppProjectOptions } from './kfintech-typescript-app-project';
export interface KFinTechNestJSAppProjectOptions extends KFinTechTypeScriptAppProjectOptions {
    /**
     * Name to different files like controller, service, module.
     *
     * @default hello
     *
     */
    readonly projectName?: string;
    /**
     * Add a library to NestJS project.
     *
     * @default true
     *
     */
    readonly lib?: boolean;
    /**
     * Add provider part of contract testing to NestJS project.
     *
     * @default true
     *
     */
    readonly contract?: boolean;
}
/**
 * Create a [NestJS](https://docs.nestjs.com) TypeScript project
 * @pjid kfintech-nestjs-app
 */
export declare class KFinTechNestJSAppProject extends KFinTechTypeScriptAppProject {
    constructor(options: KFinTechNestJSAppProjectOptions);
}
