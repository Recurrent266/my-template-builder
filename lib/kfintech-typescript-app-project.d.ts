import { TypeScriptAppProject, TypeScriptProjectOptions } from 'projen/lib/typescript';
export declare enum KFinTechTSConfigBase {
    NODE_VERSION = "node18"
}
export interface KFinTechTypeScriptAppProjectOptions extends TypeScriptProjectOptions {
    readonly addDefaultBundle?: boolean;
    readonly esmSupportConfig?: boolean;
    readonly eslintFixableAsWarn?: boolean;
    readonly nodeVersion?: string;
    readonly tsconfigBase?: KFinTechTSConfigBase;
    readonly tsconfigBaseDev?: KFinTechTSConfigBase;
    readonly tsconfigBaseStrictest?: boolean;
}
/**
 * Create a [TypeScriptAppProject]
 * @pjid kfintech-typescript-app
 */
export declare class KFinTechTypeScriptAppProject extends TypeScriptAppProject {
    constructor(options: KFinTechTypeScriptAppProjectOptions);
}
