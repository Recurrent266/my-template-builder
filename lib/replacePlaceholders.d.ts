declare const replaceInFile: (filePath: string, replacements: Array<[RegExp, string]>) => void;
declare const renameFile: (filePath: string, newFilePath: string) => void;
declare const deleteFolder: (folderPath: string) => void;
declare const copyFilesAndFolders: (sourcePath: string, destinationPath: string, projectName: string) => void;
export { replaceInFile, renameFile, deleteFolder, copyFilesAndFolders };
