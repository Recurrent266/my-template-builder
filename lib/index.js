"use strict";
// Â© 2024 Amazon Web Services, Inc. or its affiliates.
// All Rights Reserved. This AWS Content is provided subject to the terms of
// the AWS Customer Agreement available at <http://aws.amazon.com/agreement>
// or other written agreement between Customer and either
// Amazon Web Services, Inc. or Amazon Web Service EMEA SARL or both.
// Copyright 2024 Amazon.com and its affiliates; all rights reserved.
// This file is Amazon Web Services Content and may not be duplicated
// or distributed without permission.
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
Object.defineProperty(exports, "__esModule", { value: true });
__exportStar(require("./kfintech-typescript-app-project"), exports);
__exportStar(require("./kfintech-nestjs-app-project"), exports);
__exportStar(require("./utils/replace-placeholders"), exports);
//# sourceMappingURL=data:application/json;base64,eyJ2ZXJzaW9uIjozLCJmaWxlIjoiaW5kZXguanMiLCJzb3VyY2VSb290IjoiIiwic291cmNlcyI6WyIuLi9zcmMvaW5kZXgudHMiXSwibmFtZXMiOltdLCJtYXBwaW5ncyI6IjtBQUFBLHNEQUFzRDtBQUN0RCw0RUFBNEU7QUFDNUUsNEVBQTRFO0FBQzVFLHlEQUF5RDtBQUN6RCxxRUFBcUU7QUFDckUscUVBQXFFO0FBQ3JFLHFFQUFxRTtBQUNyRSxxQ0FBcUM7Ozs7Ozs7Ozs7Ozs7Ozs7QUFFckMsb0VBQWtEO0FBQ2xELGdFQUE4QztBQUM5QywrREFBNkMiLCJzb3VyY2VzQ29udGVudCI6WyIvLyDCqSAyMDI0IEFtYXpvbiBXZWIgU2VydmljZXMsIEluYy4gb3IgaXRzIGFmZmlsaWF0ZXMuXG4vLyBBbGwgUmlnaHRzIFJlc2VydmVkLiBUaGlzIEFXUyBDb250ZW50IGlzIHByb3ZpZGVkIHN1YmplY3QgdG8gdGhlIHRlcm1zIG9mXG4vLyB0aGUgQVdTIEN1c3RvbWVyIEFncmVlbWVudCBhdmFpbGFibGUgYXQgPGh0dHA6Ly9hd3MuYW1hem9uLmNvbS9hZ3JlZW1lbnQ+XG4vLyBvciBvdGhlciB3cml0dGVuIGFncmVlbWVudCBiZXR3ZWVuIEN1c3RvbWVyIGFuZCBlaXRoZXJcbi8vIEFtYXpvbiBXZWIgU2VydmljZXMsIEluYy4gb3IgQW1hem9uIFdlYiBTZXJ2aWNlIEVNRUEgU0FSTCBvciBib3RoLlxuLy8gQ29weXJpZ2h0IDIwMjQgQW1hem9uLmNvbSBhbmQgaXRzIGFmZmlsaWF0ZXM7IGFsbCByaWdodHMgcmVzZXJ2ZWQuXG4vLyBUaGlzIGZpbGUgaXMgQW1hem9uIFdlYiBTZXJ2aWNlcyBDb250ZW50IGFuZCBtYXkgbm90IGJlIGR1cGxpY2F0ZWRcbi8vIG9yIGRpc3RyaWJ1dGVkIHdpdGhvdXQgcGVybWlzc2lvbi5cblxuZXhwb3J0ICogZnJvbSAnLi9rZmludGVjaC10eXBlc2NyaXB0LWFwcC1wcm9qZWN0JztcbmV4cG9ydCAqIGZyb20gJy4va2ZpbnRlY2gtbmVzdGpzLWFwcC1wcm9qZWN0JztcbmV4cG9ydCAqIGZyb20gJy4vdXRpbHMvcmVwbGFjZS1wbGFjZWhvbGRlcnMnO1xuIl19