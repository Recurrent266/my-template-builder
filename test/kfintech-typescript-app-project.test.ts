// Â© 2024 Amazon Web Services, Inc. or its affiliates.
// All Rights Reserved. This AWS Content is provided subject to the terms of
// the AWS Customer Agreement available at <http://aws.amazon.com/agreement>
// or other written agreement between Customer and either
// Amazon Web Services, Inc. or Amazon Web Service EMEA SARL or both.
// Copyright 2024 Amazon.com and its affiliates; all rights reserved.
// This file is Amazon Web Services Content and may not be duplicated
// or distributed without permission.

import { Testing } from 'projen/lib/testing';
import { KFinTechTypeScriptAppProject } from '../src/';

const ORIGINAL_PROCESS = process;
beforeEach(() => {
  global.process = {
    ...ORIGINAL_PROCESS,
    versions: { ...ORIGINAL_PROCESS.versions, node: '18.18.2' },
  };
});
afterEach(() => {
  global.process = ORIGINAL_PROCESS;
});

test('KFinTechTypeScriptAppProject has required configuration', () => {
  const PROJECT = new KFinTechTypeScriptAppProject({
    name: 'test',
    defaultReleaseBranch: 'main',
    // default settings
  });
  // synthSnapshot(project, { parseJson: true });
  const SNAPSHOT = Testing.synth(PROJECT);

  expect(SNAPSHOT['tsconfig.json']).toMatchSnapshot();
  expect(SNAPSHOT['tsconfig.dev.json']).toMatchSnapshot();
  expect(SNAPSHOT['package.json']).toMatchSnapshot();
  expect(SNAPSHOT['.vscode/settings.json']).toMatchSnapshot();
  expect(SNAPSHOT['.vscode/extensions.json']).toMatchSnapshot();
  expect(SNAPSHOT['.eslintrc.json']).toMatchSnapshot();
  expect(SNAPSHOT['.nvmrc']).toMatchSnapshot();
  expect(SNAPSHOT['src/index.ts']).toBeDefined();
  expect(SNAPSHOT['src/hello.ts']).toBeDefined();
  expect(SNAPSHOT['test/hello.test.ts']).toBeDefined();
  /* eslint-disable */
  const TASKS = SNAPSHOT['.projen/tasks.json']?.tasks;
  const BUNDLE_TASK = TASKS['bundle:index'];
  const BUNDLE_COMMAND = BUNDLE_TASK.steps[0]?.exec;
  /* eslint-disable */
  expect(BUNDLE_COMMAND).toContain('--format=esm');
});

test('KFinTechTypeScriptAppProject no vscode settings generated', () => {
  const PROJECT = new KFinTechTypeScriptAppProject({
    name: 'test',
    defaultReleaseBranch: 'main',
    vscode: false,
    // default settings
  });
  // synthSnapshot(project, { parseJson: true });
  const SNAPSHOT = Testing.synth(PROJECT);

  expect(SNAPSHOT['.vscode/settings.json']).toBeUndefined();
  expect(SNAPSHOT['.vscode/extensions.json']).toBeUndefined();
});

test('KFinTechTypeScriptAppProject accepts eslintFixableAsWarn=false', () => {
  const PROJECT = new KFinTechTypeScriptAppProject({
    name: 'test',
    defaultReleaseBranch: 'main',
    eslintFixableAsWarn: false,
    // default settings
  });
  // synthSnapshot(project, { parseJson: true });
  const SNAPSHOT = Testing.synth(PROJECT);

  expect(SNAPSHOT['.eslintrc.json']).toMatchSnapshot('.eslintrc.json');
});
