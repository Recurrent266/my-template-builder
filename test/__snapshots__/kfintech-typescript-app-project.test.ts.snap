// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`KFinTechTypeScriptAppProject accepts eslintFixableAsWarn=false: .eslintrc.json 1`] = `
{
  "env": {
    "jest": true,
    "node": true,
  },
  "extends": [
    "plugin:import/typescript",
    "plugin:prettier/recommended",
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
  ],
  "ignorePatterns": [
    "*.js",
    "*.d.ts",
    "node_modules/",
    "*.generated.ts",
    "coverage",
    "!.projenrc.ts",
    "!projenrc/**/*.ts",
  ],
  "overrides": [
    {
      "files": [
        ".projenrc.ts",
      ],
      "rules": {
        "max-len": "off",
      },
    },
    {
      "files": [
        ".projenrc.ts",
      ],
      "rules": {
        "@typescript-eslint/no-require-imports": "off",
        "import/no-extraneous-dependencies": "off",
      },
    },
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2018,
    "project": "./tsconfig.dev.json",
    "sourceType": "module",
  },
  "plugins": [
    "@typescript-eslint",
    "import",
    "prettier",
  ],
  "root": true,
  "rules": {
    "@typescript-eslint/array-type": [
      "error",
      {
        "default": "generic",
      },
    ],
    "@typescript-eslint/ban-ts-comment": [
      "error",
      {
        "ts-expect-error": true,
        "ts-ignore": true,
      },
    ],
    "@typescript-eslint/member-ordering": [
      "error",
      {
        "default": [
          "public-static-field",
          "public-static-method",
          "protected-static-field",
          "protected-static-method",
          "private-static-field",
          "private-static-method",
          "field",
          "constructor",
          "method",
        ],
      },
    ],
    "@typescript-eslint/naming-convention": [
      "error",
      {
        "format": [
          "camelCase",
        ],
        "prefix": [
          "is",
          "should",
          "has",
          "can",
          "did",
          "will",
        ],
        "selector": "variable",
        "types": [
          "boolean",
        ],
      },
      {
        "format": [
          "camelCase",
        ],
        "selector": "function",
      },
      {
        "format": [
          "PascalCase",
        ],
        "selector": "class",
      },
      {
        "format": [
          "PascalCase",
        ],
        "selector": "interface",
      },
      {
        "format": [
          "PascalCase",
        ],
        "selector": "typeAlias",
      },
      {
        "custom": {
          "match": true,
          "regex": "^T[A-Z]",
        },
        "format": [
          "PascalCase",
        ],
        "selector": "typeParameter",
      },
    ],
    "@typescript-eslint/no-explicit-any": "error",
    "@typescript-eslint/no-floating-promises": [
      "error",
    ],
    "@typescript-eslint/no-inferrable-types": [
      "error",
      {
        "ignoreParameters": true,
        "ignoreProperties": true,
      },
    ],
    "@typescript-eslint/no-non-null-assertion": "error",
    "@typescript-eslint/no-require-imports": [
      "error",
    ],
    "@typescript-eslint/no-shadow": [
      "error",
    ],
    "@typescript-eslint/no-unused-vars": "error",
    "@typescript-eslint/return-await": [
      "error",
    ],
    "comma-dangle": [
      "error",
      "always-multiline",
    ],
    "dot-notation": [
      "error",
    ],
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": [
          "**/test/**",
          "**/build-tools/**",
          ".projenrc.ts",
          "projenrc/**/*.ts",
        ],
        "optionalDependencies": false,
        "peerDependencies": true,
      },
    ],
    "import/no-unresolved": [
      "error",
    ],
    "import/order": [
      "warn",
      {
        "alphabetize": {
          "caseInsensitive": true,
          "order": "asc",
        },
        "groups": [
          "builtin",
          "external",
        ],
      },
    ],
    "indent": [
      "error",
      2,
    ],
    "key-spacing": [
      "error",
    ],
    "max-lines": [
      "error",
      300,
    ],
    "no-bitwise": [
      "error",
    ],
    "no-console": "warn",
    "no-duplicate-imports": [
      "error",
    ],
    "no-multiple-empty-lines": [
      "error",
    ],
    "no-return-await": [
      "off",
    ],
    "no-shadow": [
      "off",
    ],
    "no-trailing-spaces": [
      "error",
    ],
    "prettier/prettier": "error",
    "quotes": [
      "error",
      "single",
      {
        "allowTemplateLiterals": true,
        "avoidEscape": true,
      },
    ],
    "semi": [
      "error",
      "always",
    ],
  },
  "settings": {
    "import/parsers": {
      "@typescript-eslint/parser": [
        ".ts",
        ".tsx",
      ],
    },
    "import/resolver": {
      "node": {},
      "typescript": {
        "alwaysTryTypes": true,
        "project": "./tsconfig.dev.json",
      },
    },
  },
}
`;

exports[`KFinTechTypeScriptAppProject has required configuration 1`] = `
{
  "compilerOptions": {
    "esModuleInterop": true,
    "exactOptionalPropertyTypes": true,
    "forceConsistentCasingInFileNames": true,
    "isolatedModules": true,
    "moduleResolution": "node16",
    "noEmit": true,
    "noFallthroughCasesInSwitch": true,
    "noImplicitOverride": true,
    "noImplicitReturns": true,
    "noPropertyAccessFromIndexSignature": true,
    "noUncheckedIndexedAccess": true,
    "noUnusedLocals": true,
    "noUnusedParameters": true,
    "outDir": "lib",
    "rootDir": "src",
    "skipLibCheck": true,
    "strict": true,
  },
  "exclude": [],
  "extends": "@tsconfig/node18/tsconfig.json",
  "include": [
    "src/**/*.ts",
  ],
  "ts-node": {
    "esm": true,
    "experimentalSpecifierResolution": "node",
    "preferTsExts": false,
  },
}
`;

exports[`KFinTechTypeScriptAppProject has required configuration 2`] = `
{
  "compilerOptions": {
    "moduleResolution": "node16",
    "noEmit": true,
  },
  "exclude": [
    "node_modules",
  ],
  "extends": "@tsconfig/node18/tsconfig.json",
  "include": [
    "src/**/*.ts",
    "test/**/*.ts",
    ".projenrc.ts",
    "projenrc/**/*.ts",
  ],
  "ts-node": {
    "esm": true,
    "experimentalSpecifierResolution": "node",
    "preferTsExts": false,
  },
}
`;

exports[`KFinTechTypeScriptAppProject has required configuration 3`] = `
{
  "//": "~~ Generated by projen. To modify, edit .projenrc.ts and run "npx projen".",
  "devDependencies": {
    "@tsconfig/node18": "*",
    "@types/jest": "*",
    "@types/node": "^18",
    "@typescript-eslint/eslint-plugin": "^6",
    "@typescript-eslint/parser": "^6",
    "constructs": "^10.0.0",
    "esbuild": "*",
    "eslint": "^8",
    "eslint-config-prettier": "*",
    "eslint-import-resolver-typescript": "*",
    "eslint-plugin-import": "*",
    "eslint-plugin-prettier": "*",
    "jest": "*",
    "jest-junit": "^15",
    "prettier": "*",
    "projen": ">=0.79.23",
    "ts-jest": "*",
    "ts-node": "*",
    "typescript": "*",
  },
  "jest": {
    "clearMocks": true,
    "collectCoverage": true,
    "coverageDirectory": "coverage",
    "coveragePathIgnorePatterns": [
      "/node_modules/",
    ],
    "coverageReporters": [
      "json",
      "lcov",
      "clover",
      "cobertura",
      "text",
    ],
    "moduleNameMapper": {
      "^(\\.{1,2}/.*)\\.m?js$": "$1",
    },
    "reporters": [
      "default",
      [
        "jest-junit",
        {
          "outputDirectory": "test-reports",
        },
      ],
    ],
    "testMatch": [
      "<rootDir>/src/**/__tests__/**/*.ts?(x)",
      "<rootDir>/@(test|src)/**/*(*.)@(spec|test).ts?(x)",
    ],
    "testPathIgnorePatterns": [
      "/node_modules/",
    ],
    "transform": {
      "^.+\\.m?[tj]sx?$": [
        "ts-jest",
        {
          "tsconfig": "tsconfig.dev.json",
          "useESM": true,
        },
      ],
    },
    "watchPathIgnorePatterns": [
      "/node_modules/",
    ],
  },
  "license": "Apache-2.0",
  "name": "test",
  "publishConfig": {
    "access": "public",
  },
  "scripts": {
    "build": "npx projen build",
    "bundle": "npx projen bundle",
    "bundle:index": "npx projen bundle:index",
    "bundle:index:watch": "npx projen bundle:index:watch",
    "clobber": "npx projen clobber",
    "compile": "npx projen compile",
    "default": "npx projen default",
    "eject": "npx projen eject",
    "eslint": "npx projen eslint",
    "package": "npx projen package",
    "post-compile": "npx projen post-compile",
    "post-upgrade": "npx projen post-upgrade",
    "pre-compile": "npx projen pre-compile",
    "projen": "npx projen",
    "test": "npx projen test",
    "test:update": "npx projen test:update",
    "test:watch": "npx projen test:watch",
    "upgrade": "npx projen upgrade",
    "watch": "npx projen watch",
  },
  "type": "module",
  "version": "0.0.0",
}
`;

exports[`KFinTechTypeScriptAppProject has required configuration 4`] = `
{
  "//": "~~ Generated by projen. To modify, edit .projenrc.ts and run "npx projen".",
  "jest.jestCommandLine": "npm test --",
  "jest.rootPath": "./",
}
`;

exports[`KFinTechTypeScriptAppProject has required configuration 5`] = `
{
  "recommendations": [
    "dbaeumer.vscode-eslint",
    "Orta.vscode-jest",
  ],
}
`;

exports[`KFinTechTypeScriptAppProject has required configuration 6`] = `
{
  "env": {
    "jest": true,
    "node": true,
  },
  "extends": [
    "plugin:import/typescript",
    "plugin:prettier/recommended",
    "eslint:recommended",
    "plugin:@typescript-eslint/recommended",
    "plugin:@typescript-eslint/recommended-requiring-type-checking",
  ],
  "ignorePatterns": [
    "*.js",
    "*.d.ts",
    "node_modules/",
    "*.generated.ts",
    "coverage",
    "!.projenrc.ts",
    "!projenrc/**/*.ts",
  ],
  "overrides": [
    {
      "files": [
        ".projenrc.ts",
      ],
      "rules": {
        "max-len": "off",
      },
    },
    {
      "files": [
        ".projenrc.ts",
      ],
      "rules": {
        "@typescript-eslint/no-require-imports": "off",
        "import/no-extraneous-dependencies": "off",
      },
    },
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "ecmaVersion": 2018,
    "project": "./tsconfig.dev.json",
    "sourceType": "module",
  },
  "plugins": [
    "@typescript-eslint",
    "import",
    "prettier",
  ],
  "root": true,
  "rules": {
    "@typescript-eslint/array-type": [
      "error",
      {
        "default": "generic",
      },
    ],
    "@typescript-eslint/ban-ts-comment": [
      "error",
      {
        "ts-expect-error": true,
        "ts-ignore": true,
      },
    ],
    "@typescript-eslint/member-ordering": [
      "error",
      {
        "default": [
          "public-static-field",
          "public-static-method",
          "protected-static-field",
          "protected-static-method",
          "private-static-field",
          "private-static-method",
          "field",
          "constructor",
          "method",
        ],
      },
    ],
    "@typescript-eslint/naming-convention": [
      "error",
      {
        "format": [
          "camelCase",
        ],
        "prefix": [
          "is",
          "should",
          "has",
          "can",
          "did",
          "will",
        ],
        "selector": "variable",
        "types": [
          "boolean",
        ],
      },
      {
        "format": [
          "camelCase",
        ],
        "selector": "function",
      },
      {
        "format": [
          "PascalCase",
        ],
        "selector": "class",
      },
      {
        "format": [
          "PascalCase",
        ],
        "selector": "interface",
      },
      {
        "format": [
          "PascalCase",
        ],
        "selector": "typeAlias",
      },
      {
        "custom": {
          "match": true,
          "regex": "^T[A-Z]",
        },
        "format": [
          "PascalCase",
        ],
        "selector": "typeParameter",
      },
    ],
    "@typescript-eslint/no-explicit-any": "error",
    "@typescript-eslint/no-floating-promises": [
      "error",
    ],
    "@typescript-eslint/no-inferrable-types": [
      "error",
      {
        "ignoreParameters": true,
        "ignoreProperties": true,
      },
    ],
    "@typescript-eslint/no-non-null-assertion": "error",
    "@typescript-eslint/no-require-imports": [
      "error",
    ],
    "@typescript-eslint/no-shadow": [
      "error",
    ],
    "@typescript-eslint/no-unused-vars": "error",
    "@typescript-eslint/return-await": [
      "error",
    ],
    "comma-dangle": [
      "error",
      "always-multiline",
    ],
    "dot-notation": [
      "warn",
    ],
    "import/no-extraneous-dependencies": [
      "error",
      {
        "devDependencies": [
          "**/test/**",
          "**/build-tools/**",
          ".projenrc.ts",
          "projenrc/**/*.ts",
        ],
        "optionalDependencies": false,
        "peerDependencies": true,
      },
    ],
    "import/no-unresolved": [
      "error",
    ],
    "import/order": [
      "warn",
      {
        "alphabetize": {
          "caseInsensitive": true,
          "order": "asc",
        },
        "groups": [
          "builtin",
          "external",
        ],
      },
    ],
    "indent": [
      "error",
      2,
    ],
    "key-spacing": [
      "warn",
    ],
    "max-lines": [
      "error",
      300,
    ],
    "no-bitwise": [
      "error",
    ],
    "no-console": "warn",
    "no-duplicate-imports": [
      "error",
    ],
    "no-multiple-empty-lines": [
      "warn",
    ],
    "no-return-await": [
      "off",
    ],
    "no-shadow": [
      "off",
    ],
    "no-trailing-spaces": [
      "warn",
    ],
    "prettier/prettier": [
      "warn",
    ],
    "quotes": [
      "error",
      "single",
      {
        "allowTemplateLiterals": true,
        "avoidEscape": true,
      },
    ],
    "semi": [
      "error",
      "always",
    ],
  },
  "settings": {
    "import/parsers": {
      "@typescript-eslint/parser": [
        ".ts",
        ".tsx",
      ],
    },
    "import/resolver": {
      "node": {},
      "typescript": {
        "alwaysTryTypes": true,
        "project": "./tsconfig.dev.json",
      },
    },
  },
}
`;

exports[`KFinTechTypeScriptAppProject has required configuration 7`] = `"v18.18.2"`;
